components/Charts/GeneralChart.tsx

```diff
@@ -0,0 +1,138 @@
+} from "@ironfish/ui-kit";
+
+import { Metric } from "types";
+
+export interface GeneralChartProps {
+  data: Metric[];
+  yAccessor: (metric: Metric) => number;
+  marginLeft?: number;
+  leftAxisFormatter?: TickFormatter<ScaleInput<AnyD3Scale>>;
+}
+
+const GeneralChart: FC<GeneralChartProps> = ({
+  data,
+  yAccessor,
+  marginLeft = 50,
+  leftAxisFormatter,
+}) => {
+  const $colors = useColorModeValue(
+    {
+      chartTheme: buildChartTheme({
+        backgroundColor: NAMED_COLORS.WHITE,
+        colors: ["url(#area)", NAMED_COLORS.LIGHT_BLUE],
+        gridColor: NAMED_COLORS.LIGHT_GREY,
+        gridColorDark: NAMED_COLORS.LIGHT_GREY,
+        svgLabelSmall: {
+          fill: NAMED_COLORS.GREY,
+          fontSize: "12px",
+          fontFamily: "favorit-regular",
+        },
+        svgLabelBig: { fill: NAMED_COLORS.BLACK },
+        tickLength: 0,
+      }),
+      bottomAxisLabelColor: NAMED_COLORS.BLACK,
+    },
+    {
+      chartTheme: buildChartTheme({
+        backgroundColor: NAMED_COLORS.DARKER_GREY,
+        colors: ["url(#area)", NAMED_COLORS.LIGHT_BLUE],
+        gridColor: NAMED_COLORS.DARK_GREY,
+        gridColorDark: NAMED_COLORS.DARK_GREY,
+        svgLabelSmall: {
+          fill: NAMED_COLORS.PALE_GREY,
+          fontSize: "0.75rem",
+          fontFamily: "ABC Favorit Trial",
+        },
+        svgLabelBig: { fill: NAMED_COLORS.WHITE },
+        tickLength: 0,
+      }),
+      bottomAxisLabelColor: NAMED_COLORS.WHITE,
+    }
+  );
+  const $bottomAxisTicks = useBreakpointValue({ base: 4, md: 10 });
+  const xAccessor = (metric: Metric): Date => new Date(metric.date);
+
+  return (
+    <XYChart
+      theme={$colors.chartTheme}
+      xScale={{ type: "utc" }}
+      yScale={{ type: "linear" }}
+      height={340}
+      margin={{ left: marginLeft, right: 0, top: 20, bottom: 35 }}
+    >
+      <LinearGradient from={NAMED_COLORS.LIGHT_BLUE} to="#2C72FF00" id="area" />
+      <Grid numTicks={8} columns={false} />
+      <AreaSeries
+        dataKey="transactions"
+        data={data}
+        xAccessor={xAccessor}
+        yAccessor={yAccessor}
+      />
+      <LineSeries
+        dataKey="transactionsLine"
+        data={data}
+        xAccessor={xAccessor}
+        yAccessor={yAccessor}
+        strokeWidth={2}
+      />
+      <Axis
+        orientation="left"
+        numTicks={8}
+        tickTransform="translate(-8, 0)"
+        hideAxisLine
+        tickFormat={leftAxisFormatter}
+      />
+      <Axis
+        orientation="bottom"
+        numTicks={$bottomAxisTicks}
+        hideTicks
+        tickLabelProps={() => ({
+          fill: $colors.bottomAxisLabelColor,
+          fontSize: "0.75rem",
+          fontFamily: "ABC Favorit Trial",
+        })}
+        tickTransform="translate(0, 18)"
+        hideAxisLine
+        tickFormat={(d) => {
+          const day = d.getDate() > 1 ? `. ${d.getDate()}` : "";
+          return `${d.toLocaleString("default", { month: "short" })}${day}`;
```

Comment:
Let's start using `date-fns` instead of this. I filed a [ticket](https://linear.app/ironfish/issue/IRO-1976/avoid-using-native-date-methods-and-start-using-date-fns) so this isn't a blocker, but I keep seeing native Date methods and they're a PITA for maintenance